### PR TITLE
Sequence to optimizations

### PR DESCRIPTION
**Pull request by anderspitman**
Minimal speedup (4.44us -> 4.03us) Not sure if its worth it.

### CODE REVIEW

File: skbio/sequence/_sequence.py [hunk 1]
```diff
@@ -1860,7 +1860,7 @@ def iter_contiguous(self, included, min_length=1, invert=False):
             if len(r) >= min_length:
                 yield r
 
-    def _to(self, **kwargs):
+    def _to(self, sequence=None, metadata=None, positional_metadata=None):
         """Return a copy of the current biological sequence.
```
Comment (jairideout) [inline skbio/sequence/_sequence.py]: Can you update (or just remove... it's private) the outdated docs?

**Comment by jairideout** [comment]
One comment and this is ready for merge!

**Comment by anderspitman** [comment]
Tabled pending more in depth investigation of current performance.

**Comment by anderspitman** [comment]
Results [here](https://github.com/anderspitman/jupyter_notebooks/blob/62eb5bbf4c6fd096873c22a70d2c5c0b1d6529ca/iter_kmers_bench.ipynb).

**Comment by ebolyen** [comment]
We are holding off on this for now.

**Pull request merged**